lib/rules/disable-enable-pair.js

```diff
@@ -39,35 +39,35 @@ module.exports = {
     create(context) {
         const allowWholeFile =
             context.options[0] && context.options[0].allowWholeFile
-        const sourceCode = context.getSourceCode()
-        const disabledArea = DisabledArea.get(sourceCode)
+        const disabledArea = getDisabledArea(context)
 
-        return {
-            Program(node) {
-                if (allowWholeFile && node.body.length === 0) {
-                    return
-                }
+        /** @type {import('@eslint/core').TextSourceCode} */
+        const sourceCode = context.sourceCode || context.getSourceCode()
 
-                for (const area of disabledArea.areas) {
-                    if (area.end != null) {
-                        continue
-                    }
-                    if (
-                        allowWholeFile &&
-                        utils.lte(area.start, node.loc.start)
-                    ) {
-                        continue
-                    }
+        const firstToken =
+            sourceCode.ast && sourceCode.ast.tokens && sourceCode.ast.tokens[0]
 
-                    context.report({
-                        loc: utils.toRuleIdLocation(area.comment, area.ruleId),
-                        messageId: area.ruleId
-                            ? "missingRulePair"
-                            : "missingPair",
-                        data: area,
-                    })
-                }
-            },
+        if (allowWholeFile && !firstToken) {
+            return {}
+        }
+
+        for (const area of disabledArea.areas) {
+            if (area.end != null) {
+                continue
+            }
+            if (
+                allowWholeFile &&
+                utils.lte(area.start, utils.getLoc(context, firstToken).start)
+            ) {
+                continue
+            }
+
+            context.report({
+                loc: utils.toRuleIdLocation(context, area.comment, area.ruleId),
+                messageId: area.ruleId ? "missingRulePair" : "missingPair",
+                data: area,
+            })
         }
```

Comment:
```suggestion
        disabledArea.areas
            .filter((area) => area.end === null || !(allowWholeFile && utils.lte(area.start, utils.getLoc(context, firstToken).start)))
            .forEach((area) => context.report({
                loc: utils.toRuleIdLocation(context, area.comment, area.ruleId),
                messageId: area.ruleId ? "missingRulePair" : "missingPair",
                data: area,
            }))
        }
```

Reply:
This part respects the original implementation, and I think it's better to make a separate PR if we want to change it.